spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -152,31 +159,30 @@ public static Expression convert(Predicate predicate) {
           }
 
         case EQ: // used for both eq and null-safe-eq
-          Object value;
-          String columnName;
-          if (isRef(leftChild(predicate)) && isLiteral(rightChild(predicate))) {
-            columnName = SparkUtil.toColumnName(leftChild(predicate));
-            value = convertLiteral(rightChild(predicate));
-          } else if (isRef(rightChild(predicate)) && isLiteral(leftChild(predicate))) {
-            columnName = SparkUtil.toColumnName(rightChild(predicate));
-            value = convertLiteral(leftChild(predicate));
-          } else {
+          PredicateChildren<Object> eqChildren = predicateChildren(predicate);
+          if (eqChildren == null) {
             return null;
           }
 
-          if (predicate.name().equals(EQ)) {
-            // comparison with null in normal equality is always null. this is probably a mistake.
+          if (EQ.equals(predicate.name())) {
+            // comparison with null in normal equality is always null. this is probably a
+            // mistake.
```

Comment:
Why did the formatting of this line change? It looks like it doesn't need to.

Reply:
Reverted